### PR TITLE
[fix](auth) fix use  database stmt access  unauthorized  catalog

### PR DESCRIPTION
**Pull request by vinlee19**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
       For example, suppose there is a database named "testdb" in the internal catalog. Meanwhile, there is also a database with the same name in the external catalog. However, you only have the right to access the internal catalog. When you use the "external_catalog.testdb" statement, there is a possibility that you can access the external catalog as well.
Perhaps the issue only has a minimal effect on users. 

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Thearas** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by vinlee19** [comment]
run buildall

File: regression-test/suites/external_table_p0/nereids_commands/test_use_database_stmt.groovy [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        sql """CREATE USER ${user}@'%' IDENTIFIED BY '${pwd}';"""
+        sql """GRANT SELECT_PRIV ON ${internal_catalog}.*.* TO '${user}'@'%';"""
+
+        connect(user, pwd, url) {
+            try {
+                sql """switch internal"""
+                sql """use ${internal_db_name}"""
+                qt_sql """select current_catalog()"""
+                sql """use ${catalog_name}.${ex_db_name}"""
+                Assert.fail("Access denied for user '${user}' to database '${ex_db_name}'");
```
Comment (zddr) [inline regression-test/suites/external_table_p0/nereids_commands/test_use_database_stmt.groovy:65]: u can write it like this:
test {
          sql """
             sql """use ${catalog_name}.${ex_db_name}"""
          """
          exception "Access denied for user '${user}'"
    }

**Review by zddr — APPROVED** [review]

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run p0

**Review by LiBinfeng-01 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**